src/SPC/builder/unix/UnixBuilderBase.php

```diff
@@ -210,21 +211,71 @@ protected function sanityCheck(int $build_target): void
     /**
      * Deploy the binary file to the build bin path.
      *
-     * @param int $type Type integer, one of BUILD_TARGET_CLI, BUILD_TARGET_MICRO, BUILD_TARGET_FPM
+     * @param int $type Type integer, one of BUILD_TARGET_CLI, BUILD_TARGET_MICRO, BUILD_TARGET_FPM, BUILD_TARGET_CGI, BUILD_TARGET_FRANKENPHP
      */
-    protected function deployBinary(int $type): bool
+    protected function deployBinary(int $type): void
     {
+        FileSystem::createDir(BUILD_BIN_PATH);
+        $copy_files = [];
         $src = match ($type) {
             BUILD_TARGET_CLI => SOURCE_PATH . '/php-src/sapi/cli/php',
             BUILD_TARGET_MICRO => SOURCE_PATH . '/php-src/sapi/micro/micro.sfx',
             BUILD_TARGET_FPM => SOURCE_PATH . '/php-src/sapi/fpm/php-fpm',
             BUILD_TARGET_CGI => SOURCE_PATH . '/php-src/sapi/cgi/php-cgi',
+            BUILD_TARGET_FRANKENPHP => BUILD_BIN_PATH . '/frankenphp',
             default => throw new SPCInternalException("Deployment does not accept type {$type}"),
         };
-        logger()->info('Deploying ' . $this->getBuildTypeName($type) . ' file');
-        FileSystem::createDir(BUILD_BIN_PATH);
-        shell()->exec('cp ' . escapeshellarg($src) . ' ' . escapeshellarg(BUILD_BIN_PATH));
-        return true;
+        $no_strip_option = (bool) $this->getOption('no-strip', false);
+        $upx_option = (bool) $this->getOption('with-upx-pack', false);
+
+        // Generate debug symbols if needed
+        $copy_files[] = $src;
+        if (!$no_strip_option && PHP_OS_FAMILY === 'Darwin') {
+            shell()
+                ->exec("dsymutil -f {$src}") // generate .dwarf file
+                ->exec("strip -S {$src}"); // strip unneeded symbols
+            $copy_files[] = "{$src}.dwarf";
+        } elseif (!$no_strip_option && PHP_OS_FAMILY === 'Linux') {
+            shell()
+                ->exec("objcopy --only-keep-debug {$src} {$src}.debug") // extract debug symbols
+                ->exec("objcopy --strip-debug --add-gnu-debuglink={$src}.debug {$src}") // link debug symbols
+                ->exec("strip --strip-unneeded {$src}"); // strip unneeded symbols
```

Comment:
--strip-debug is unnecessary here, --strip-unneeded already catches it right after